### PR TITLE
Update Saburo

### PR DESCRIPTION
**Pull request by tmagicturtle**

### CODE REVIEW

File: data/mods/ssb/moves.js [hunk 1]
```diff
@@ -3285,42 +3285,56 @@ let BattleMovedex = {
...
+	magmaore: {
 		accuracy: true,
 		basePower: 0,
 		category: "Status",
-		desc: "Raises the user's Attack and Speed by 1 stage. Has a 10% chance to summon either Reflect or Light Screen.",
-		shortDesc: "Atk, Spe +1; 10% chance to set one screen.",
-		id: "soulbend",
-		name: "Soulbend",
+		desc: "Sets up a hazard on the opposing side of the field, damaging each opposing Pokemon that switches in. Fails if the effect is already active on the opposing side. Foes lose 1/16 of their maximum HP, rounded down. Has a 10% chance to freeze the foe. Causes hail on the field Can be removed from the opposing side if any opposing Pokemon uses Rapid Spin or Defog successfully, or is hit by Defog.",
+		shortDesc: "Hurts foes on switch-in; 10% freeze; sets hail.",
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: uh... I don't think fire freezes things.

File: data/mods/ssb/moves.js [hunk 2]
```diff
@@ -3285,42 +3285,56 @@ let BattleMovedex = {
...
 		priority: 0,
-		flags: {snatch: 1, mirror: 1},
+		flags: {reflectable: 1},
 		onTryMove() {
 			this.attrLastMove('[still]');
 		},
 		onPrepareHit(target, source) {
-			this.add('-anim', source, "Refresh", source);
-			this.add('-anim', source, "Geomancy", source);
+			this.add('-anim', source, 'Stealth Rock', target);
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js:3295]: Missing eruption animation.

File: data/mods/ssb/moves.js [hunk 3]
```diff
@@ -3285,42 +3285,56 @@ let BattleMovedex = {
...
+			onStart(side) {
+				this.add('-message', 'Molten lava appears on the field!');
+			},
+			onSwitchIn(pokemon) {
+				let typeMod = this.clampIntRange(pokemon.runEffectiveness(this.getActiveMove('stealthrock')), -6, 6);
+				this.damage(pokemon.maxhp * Math.pow(2, typeMod) / 8);
+				if (!(pokemon.hp <= 0 || pokemon.fainted) && this.random(10) < 4) {
+					pokemon.trySetStatus('brn');
+					// @ts-ignore
+					pokemon.magmaOre = true;
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Isn't there a `pokemon.m` value we can use for flags like this?

File: data/mods/ssb/moves.js (hunk 1 above)
Comment (tmagicturtle) [reply data/mods/ssb/moves.js]: Whoops. Used another move as a template for this and forgot to change the description.

File: data/mods/ssb/moves.js (hunk 3 above)
Comment (tmagicturtle) [reply data/mods/ssb/moves.js]: Do I just use that as pokemon.m.magmaOre?

**Comment by tmagicturtle** [comment]
Marking it here that I still need to modify Defog and Rapid Spin to remove Magma Ore.

File: data/mods/ssb/moves.js (hunk 3 above)
Comment (HoeenCoder) [reply data/mods/ssb/moves.js]: yes

**Comment by tmagicturtle** [comment]
I don't understand why this is failing. There aren't mixed tabs and spaces, and there's no trailing spaces.

File: data/mods/ssb/moves.js [hunk 4]
```diff
@@ -3276,42 +3276,54 @@ let BattleMovedex = {
...
+		effect: {
+			// this is a side condition
+			onStart(side) {
+				this.add('-message', 'Molten lava appears on the field!');
+			},
+			onSwitchIn(pokemon) {
+				let typeMod = this.clampIntRange(pokemon.runEffectiveness(this.getActiveMove('stealthrock')), -6, 6);
+				this.damage(pokemon.maxhp * Math.pow(2, typeMod) / 8);
+				if (!(pokemon.hp <= 0 || pokemon.fainted) && this.random(10) < 4) {
+					pokemon.trySetStatus('brn');
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Only set the magmaore flag if this works.

File: data/mods/ssb/moves.js [hunk 5]
```diff
@@ -3276,42 +3276,54 @@ let BattleMovedex = {
...
+			this.add('-anim', source, 'Stealth Rock', target);
+			this.add('-anim', source, 'Eruption', target);
 		},
-		boosts: {
-			atk: 1,
-			spe: 1,
+		sideCondition: 'magmaore',
+		self: {
+			onHit(pokemon) {
+				pokemon.addVolatile('magmaore2');
```
Comment (HoeenCoder) [inline data/mods/ssb/moves.js]: Add a check to make sure that this only gets applied to saburo.

**Review by HoeenCoder — COMMENTED** [review]
Yes there were issues with spacing, I fixed them.

**Comment by tmagicturtle** [comment]
> Yes there were issues with spacing, I fixed them.

Thanks. It was giving me a completely different line so I was completely lost where the spacing issue was.

**Pull request merged**